### PR TITLE
Initialize the repo with a barebones project and infrastructure.

### PR DESCRIPTION
**Pull request by tannergooding**
This adds the initial files for a barebones project and the required repo infrastructure. It pulls down novelrt via vcpkg and renders a blank screen.

image

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,54 @@
+find_package(Freetype REQUIRED)
+find_package(glad CONFIG REQUIRED)
+find_package(glfw3 CONFIG REQUIRED)
+find_package(glm CONFIG REQUIRED)
+find_package(libpng CONFIG REQUIRED)
+find_package(LibSndFile CONFIG REQUIRED)
+find_package(NovelRT CONFIG REQUIRED)
+find_package(OpenAL CONFIG REQUIRED)
+find_package(spdlog CONFIG REQUIRED)
```
Comment (RubyNova) [inline src/CMakeLists.txt:9]: Surely we should just have to `find_package NovelRT` here? As opposed to all of them?

File: src/main.cpp [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root for more information.
+
+#include <NovelRT.h>
+
+int main(int argc, char *argv[])
+{
```
Comment (RubyNova) [inline src/main.cpp]: bracing style is off here, should probably be K&R as opposed to Allman.

**Review by RubyNova — COMMENTED** [review]
LGTM bar 1 minor change and a question. Once those are resolved I'll merge it.

File: src/CMakeLists.txt (hunk 1 above)
Comment (tannergooding) [reply src/CMakeLists.txt:9]: We get errors like:
```
CMake Error at artifacts/vcpkg/scripts/buildsystems/vcpkg.cmake:188 (_add_executable):
  Target "TouhouNovelRT" links to target "glad::glad" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
Call Stack (most recent call first):
  src/CMakeLists.txt:13 (add_executable)
```

I'd guess I just don't have the targets quite right and it's something I'll have to investigate more to fix.

Comment (RubyNova) [reply src/CMakeLists.txt:9]: Do you think we could get this fixed before merging? Or nah?

File: src/main.cpp (hunk 2 above)
Comment (tannergooding) [reply src/main.cpp]: Fixed. Force of habit 😄

File: src/CMakeLists.txt (hunk 1 above)
Comment (RubyNova) [reply src/CMakeLists.txt:9]: Discussed on Discord. Will resolve at a later date.

**Pull request merged**